File: rb/features.rb
# DutchCustomerData SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module DutchCustomerDataFeatures
  def self.make_feature(name)
    case name
    when "base"
      DutchCustomerDataBaseFeature.new
    when "ratelimit"
      DutchCustomerDataRatelimitFeature.new
    when "retry"
      DutchCustomerDataRetryFeature.new
    when "test"
      DutchCustomerDataTestFeature.new
    when "timeout"
      DutchCustomerDataTimeoutFeature.new
    else
      DutchCustomerDataBaseFeature.new
    end
  end
end

File: py/features.py
# DutchCustomerData SDK feature factory

from feature.base_feature import DutchCustomerDataBaseFeature
from feature.test_feature import DutchCustomerDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DutchCustomerDataBaseFeature(),
        "test": lambda: DutchCustomerDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
